pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -0,0 +1,171 @@
+package kafka
+
+import (
+	"fmt"
+)
+
+func generateConsumerSQL(start, end int64, topic, partition, consumerGroup, queueType string) string {
+	query := fmt.Sprintf(`
+WITH 
+-- Sub query for p99 calculation
+p99_query AS (
+    SELECT
+        serviceName,
+        quantile(0.99)(durationNano) / 1000000 as p99
+    FROM signoz_traces.signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+      AND timestamp <= '%d'
+      AND kind = 5
+      AND msgSystem = '%s' 
+      AND stringTagMap['messaging.destination.name'] = '%s'
+      AND stringTagMap['messaging.destination.partition.id'] = '%s'
+	  AND stringTagMap['messaging.kafka.consumer.group'] = '%s'
+    GROUP BY serviceName
+),
+
+-- Sub query for RPS calculation
+rps_query AS (
+    SELECT
+        serviceName,
+        count(*) / ((%d - %d) / 1000000000) AS rps  -- Convert nanoseconds to seconds
+    FROM signoz_traces.signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+        AND timestamp <= '%d'
+        AND kind = 5
+		AND msgSystem = '%s' 
+        AND stringTagMap['messaging.destination.name'] = '%s'
+        AND stringTagMap['messaging.destination.partition.id'] = '%s'
+		AND stringTagMap['messaging.kafka.consumer.group'] = '%s'
+    GROUP BY serviceName
+),
+
+-- Sub query for error rate calculation
+error_rate_query AS (
```

Comment:
do we want to show the rate as in `errors/s` or `error %` = `errors/total`?

Reply:
Personally, error% makes more sense to me. For example, if there were 20 errors in a minute but the processed records are 100k, the error as a percentage of the total processed tells a different story than errors/s.

Reply:
SGTM